internalserver/handler.go

```diff
@@ -0,0 +1,104 @@
+package internalserver
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/pprof"
+	"sort"
+
+	"github.com/metalmatze/signal/healthcheck"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+type Handler struct {
+	http.ServeMux
+	endpoints map[string]string
```

Comment:
Are we only really using a map to simplify the test case?

Rather than a map, I think we could directly use a slice of the type you defined later:
```go
	type endpoint struct {
		Pattern     string
		Description string
	}
```
Or even a slice of two-element arrays. The upside is less memory usage and also the order is stable so we don't need to re-sort on every index render.

Reply:
That's true. The idea of the map was to make sure to only have one endpoint ever for  a given pattern. Maybe that's already what the http.ServeMux checks already.